### PR TITLE
simulators/ethereum/engine: struct inputs and TransitionConfigurationV1 tests

### PR DESCRIPTION
**Pull request by MarcusWentz**
Testing inputs for Engine API struct parameters (including TransitionConfigurationV1):
-prefix (0x00)
-long
-short

TransitionConfigurationV1 tests check to see if CL == EL for parameters:
-TTD
-TerminalBlockNumber
-TerminalBlockHash

Note: cannot directly test Uint64/BigInt parameters due to marshaling restriction. 
Could possibly be tested with building raw packets/RLP but also should be covered by Engine API Fuzz tests.

### CODE REVIEW

File: simulators/ethereum/engine/teststruct.go [hunk 1]
```diff
@@ -230,6 +248,61 @@ func (exp *GetPayloadResponseExpectObject) ExpectTimestamp(expectedTimestamp uin
 	}
 }
 
+// TransitionConfigurationV1
+type ExchangeTransitionConfigurationResponseExpectObject struct {
```
Comment (holiman) [inline simulators/ethereum/engine/teststruct.go:252]: This type of name is pretty common in the Java world, but in go-lang, people usually tend to use shorter names. Also, the docstring is wrong -- it should always begin with the name of the struct it is describing. 

Could you please write a brief description of what this struct is/does, and then maybe we can make the name a bit briefer.

File: simulators/ethereum/engine/teststruct.go [hunk 2]
```diff
@@ -230,6 +248,34 @@ func (exp *GetPayloadResponseExpectObject) ExpectTimestamp(expectedTimestamp uin
 	}
 }
 
+// ExchangeTransitionConfigurationV1 in bytes format.
+type ExTransConfigResponseExpectObject struct {
+	*TestEnv
```
Comment (MarcusWentz) [inline simulators/ethereum/engine/teststruct.go:253]: Docstring corrected and routine name abbreviated. 

Rewriting types as bytes in this context allows test size inputs beyond the original struct data types.

File: simulators/ethereum/engine/teststruct.go [hunk 3]
```diff
@@ -230,6 +248,61 @@ func (exp *GetPayloadResponseExpectObject) ExpectTimestamp(expectedTimestamp uin
...
+func (tec *TestEngineClient) TestEngineExchangeTransitionConfigurationV1(payload *TransitionConfigurationV1) *ExchangeTransitionConfigurationResponseExpectObject {
+	response, err := tec.EngineClient.EngineExchangeTransitionConfigurationV1(tec.EngineClient.Ctx(), payload)
+	return &ExchangeTransitionConfigurationResponseExpectObject{
+		TestEnv: tec.TestEnv,
+		Response: response,
+		Error:   err,
+	}
+}
+
+type ExchangeTransitionConfigurationBytesResponseExpectObject struct {
```
[the remark this answers is not in the document]
Comment (MarcusWentz) [reply simulators/ethereum/engine/teststruct.go]: The original struct was not compatible with the modified bytes version. 

The original struct was not used for any tests, so I have removed the original version but kept the bytes version below.

**Comment by MarcusWentz** [comment]
Closing PR since Geth patch added for logsBloom https://github.com/ethereum/go-ethereum/pull/24946

**Pull request closed without merging**